backend/Tim.Backend/Providers/Database/CouchbaseRepository.cs

```diff
@@ -0,0 +1,129 @@
+        where TJsonEntity : IJsonEntity
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CouchbaseRepository{TJsonEntity}"/> class.
+        /// </summary>
+        /// <param name="dbClient">Couchbase client.</param>
+        public CouchbaseRepository(CouchbaseDbClient dbClient)
+        {
+            Logger = Log.Logger;
+
+            DatabaseClient = dbClient;
+            Collection = DatabaseClient.CollectionAsync<TJsonEntity>();
+        }
+
+        /// <summary>
+        /// Gets or sets the Couchbase db client.
+        /// </summary>
+        public CouchbaseDbClient DatabaseClient { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Couchbase collection associated with <typeparamref name="TJsonEntity"/>.
+        /// </summary>
+        public Task<ICouchbaseCollection> Collection { get; set; }
+
+        /// <summary>
+        /// Gets the logger.
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        /// <inheritdoc/>
+        public async Task AddOrUpdateItemAsync(IJsonEntity entity, TimeSpan? timeToLive = null)
+        {
+            var upsertOptions = new UpsertOptions();
+
+            if (timeToLive != null)
+            {
+                upsertOptions.Expiry(timeToLive.Value);
+            }
+
+            try
+            {
+                var collection = await Collection;
+                var upsertResult = await collection.UpsertAsync(entity.Id, entity, upsertOptions);
+            }
+            catch (Exception e)
+            {
+                Logger.Error(e, "CouchbaseDbClient client failed with error: " + e.ToString());
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<TJsonEntity> GetItemAsync(string id)
+        {
+            try
+            {
+                var collection = await Collection;
+                var getResult = await collection.GetAsync(id);
+
+                return getResult.ContentAs<TJsonEntity>();
+            }
+            catch (DocumentNotFoundException e)
+            {
+                // element not found
+                Logger.Error(e, $"Not able to find element with id {id}: " + e.ToString());
+                return default;
+            }
+            catch (Exception e)
+            {
+                Logger.Error(e, "CouchbaseDbClient client failed with error: " + e.ToString());
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<TJsonEntity>> GetItemsAsync()
+        {
+            try
+            {
+                var bucket = DatabaseClient.Bucket;
+                var scope = await bucket.DefaultScopeAsync();
+                var collectionName = CouchbaseDbClient.GetCollectionName<TJsonEntity>();
+                var queryResult = await scope.QueryAsync<TJsonEntity>($"SELECT d.* FROM {collectionName} d");
```

Comment:
I remember that couch base wanted database.scope.collection name here, not sure if you tested with just collection name?

Reply:
You're right, but they change the SDK to support querying the collection directly, which is one of the reasons I updated the library version.